### ISSUE TITLE
Windows x86 PE RTTI Analyzer not available in Minidump Dump File Loader

### ISSUE THREAD

**Issue by antoniovazquezblanco** [NationalSecurityAgency/ghidra#5214]
**Describe the bug**
Windows x86 PE RTTI Analyzer is not available for a file loaded with Dump File Loader in Minidump file format despite Visual Studio is selected as the compiler.

![imagen](https://user-images.githubusercontent.com/304193/230952135-fa1e38cc-e545-428d-b505-5da8b1350835.png)

**To Reproduce**
1. Dump a PE process in Windows with Sysinternals Process Explorer.
2. Import the dump file into Ghidra.
3. Open the imported file and launch the analyzer.
4. Windows x86 PE RTTI Analyzer is missing...

**Expected behavior**
I would expect Windows x86 PE RTTI Analyzer to be available.

**Screenshots**

See Description above.

**Attachments**
If applicable, please attach any files that caused problems or log files generated by the software.

**Environment (please complete the following information):**
 - OS: Windows & Linux tested
 - Java Version: 17.0.4.1
 - Ghidra Version: 10.3 (master branch at https://github.com/NationalSecurityAgency/ghidra/commit/85cc8ee883aad9c3bede7e62a0ffb75a1238989c)
 - Ghidra Origin: Locally built (also happening on github releases...)

**Additional context**
Launching the debugger on ghidra shows that the program is recognized as a PE file.

![imagen](https://user-images.githubusercontent.com/304193/230952858-ceec811e-8326-40a6-9617-5f9928d8d515.png)

Unfortunatelly, the compiler is set to unknown...

![imagen](https://user-images.githubusercontent.com/304193/230953008-a724b912-f714-4bbe-a2cf-3ce075d791cf.png)

This later condition leads to `isVisualStudioOrClangPe` in the `PEUtil` class to return false
![imagen](https://user-images.githubusercontent.com/304193/230953200-a1cc687b-e39b-4412-8e81-c009ab6327a9.png)

Thus disabing the `RttiAnalyzer`...

I'm learning to navigate Ghidra source code. I would like to contribute with a patch for this issue but I need guidance on where the compiler option should be set or how it should be set.

Thanks

**Comment by d-millar** [comment]
Agreed - this should be fixed.  May be a little trickier than you want to get into as I'm not sure the PE Image is available directly.  Let me take a whack at this first (unless you really really want to take it on).

**Comment by antoniovazquezblanco** [comment]
I looked into it a little bit more in detail. It seems that Windows x86 PE RTTI Analyzer depends on many other things such as having PE sections correctly labeled among other things...

As you said, PE Image doesn't seem to be available directly. That may be an issue on its own...

Thanks @d-millar!

**Comment by antoniovazquezblanco** [comment]
In my particular case, I do not have access to the original executable file.

I've modified the RTTI analizer to run and was able to recover a lot of RTTI0 structs but it has been tailor made to work on this particular file and not suitable for generalization.

I understand the complexity of the issue now and believe that this is not doable in the short term.

I lack a lot of knowledge about how loaders work and what structures does ghidra expose to downstream analyzers to be able to propose or work towards a long term solution. I was able to understand that the dump loader seems to not be able to pass data to tag PE structures...

I kind of wonder if this issue could be split into smaller ones, more manageable that I could invest time into investigating.

I will leave this issue open for you to decide if you want to discuss on how things should be made or if can be split. If you believe this cannot be done at the moment or that it is not reasonable to do so here, please feel free to close the issue on my befhalf as I now understand the complexity of the task.

Thanks

**Comment by d-millar** [comment]
@antoniovazquezblanco OK, thanks.  I was dithering on whether to mark this "Future" or "Won't support", but I think I'm changing my mind and will add an internal ticket for revisiting this at some point.  Will also give your idea of splitting it up some thought.  Unfortunately, the dump file loaders, particularly the Minidump version, are about as complicated as you can get.  It took me quite a while to navigate the memory block logic, and I had considerable help!  I do think a script to apply the RTTI Analyzer logic once a PE header had been found might be of general use.  Also perhaps, a script for scanning memory for PE objects generally.
